src/components/dialogs/parameters/dynamicsimulation/curve/dialog/model-filter.tsx

```diff
@@ -0,0 +1,310 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { FormattedMessage, IntlShape, useIntl } from 'react-intl';
+import { forwardRef, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState } from 'react';
+import { Grid, Box, Typography, Theme } from '@mui/material';
+import CheckboxSelect from '../common/checkbox-select';
+import CheckboxTreeview, { GetSelectedItemsHandle } from '../common/checkbox-treeview';
+import { lighten } from '@mui/material/styles';
+import { useSelector } from 'react-redux';
+
+import { fetchDynamicSimulationModels } from '../../../../../../services/study/dynamic-simulation';
+import { EQUIPMENT_TYPES } from '../../../../../utils/equipment-types';
+import { AppState } from 'redux/reducer';
+
+interface ModelVariableDefinitionInfos {
+    name: string;
+    unit: string;
+}
+
+interface DynamicSimulationModelBack {
+    modelName: string;
+    equipmentType: EQUIPMENT_TYPES;
+    variableDefinitions: ModelVariableDefinitionInfos[];
+    variablesSets: { name: string; variableDefinitions: ModelVariableDefinitionInfos[] }[];
+}
+
+interface DynamicSimulationModel {
+    name: string;
+    equipmentType: EQUIPMENT_TYPES;
+}
+
+export interface GetSelectedVariablesHandle {
+    api: {
+        getSelectedVariables: () => ModelVariable[];
+    };
+}
+
+interface ModelFilterProps {
+    equipmentType: EQUIPMENT_TYPES;
+}
+export type ModelVariable = {
+    id: string;
+    name: string;
+    parentId?: string;
+    variableId?: string;
+};
+
+const modelsToVariablesTree = (models: DynamicSimulationModelBack[]) => {
+    return models.reduce<Record<string, Record<string, string | Record<string, string>>>>(
```

Comment:
maybe add a type: someType = Record<string, string | Record<string, string>> and replace it in reduce models.reduce<Record<string, someType>>... easier to read

Reply:
I have proposed a new type here: https://github.com/gridsuite/gridstudy-app/pull/2418/files#diff-1210fc6bb283b8ef879fe4f9e4ca97fc79b9795f0072925de8e1d6f54b11c0cbR10
It is more semantically than using record